garak/generators/azure.py

```diff
@@ -0,0 +1,140 @@
+"""Azure OpenAI generator
+
+Supports chat + chatcompletion models. Put your API key in the AZURE_API_KEY environment variable, 
+the azure openai endpoint in the AZURE_ENDPOINT environment variable and the azure openai deployment name 
+in AZURE_DEPLOYMENT_NAME environment variable.
+
+Put the model name in either the --model_name command line parameter, or
+pass it as an argument to the Generator constructor.
+"""
+
+import random
+import os
+from typing import List, Union
+
+import openai
+
+from garak import _config
+from garak.generators.openai import OpenAICompatible, chat_models, completion_models, context_lengths
+
+# lists derived from https://learn.microsoft.com/en-us/azure/ai-services/openai/concepts/models
+# some azure openai model names should be mapped to openai names
+chat_models_openai = {
+    "gpt-4": "gpt-4-turbo-2024-04-09", 
+    "gpt-4o": "gpt-4o", # links to latest version
+    "gpt-4o-mini": "gpt-4o-mini", # links to latest version
+    "gpt-35-turbo": "gpt-3.5-turbo-0125", 
+    "gpt-35-turbo-16k": "gpt-3.5-turbo-16k", 
+    "gpt-4-32k": "gpt-4-32k", 
```

Comment:
Removed item with matching key and value.
```suggestion
    "gpt-4": "gpt-4-turbo-2024-04-09", 
    "gpt-35-turbo": "gpt-3.5-turbo-0125", 
    "gpt-35-turbo-16k": "gpt-3.5-turbo-16k", 
```

Reply:
should be fixed in the last commit